core/src/Kokkos_Complex.hpp

```diff
@@ -50,6 +50,21 @@
 
 namespace Kokkos {
 
+template<typename From, typename To, typename dummy = To>
+struct is_convertible {
```

Comment:
Does this mean the same thing as `std::is_convertible`?  If so, please just use that.  If not, I would recommend using a different name to avoid confusion.

Reply:
Also, I think what you want is that InputRealType is not complex, i.e., is neither `Kokkos::complex` nor `std::complex`.

Reply:
I removed my is_convertible. I just want to put these operator under our control but it seems to be a bad idea.